.azure-pipelines/steps/run-tests.yml

```diff
@@ -0,0 +1,24 @@
+steps:
+- task: UsePythonVersion@0
+  displayName: Use Python $(python.version)
+  inputs:
+    versionSpec: '$(python.version)'
+
+- template: install-dependencies.yml
+
+- script: tox -e py -- -m unit --junit-xml=junit/unit-test.xml
+  displayName: Tox run unit tests
+
+# Run integration tests in two groups so we will fail faster if there is a failure in the first group
```

Comment:
We can probably drop this split here. Failing fast doesn't really help all that much here.

Further, since these are being run one after the other, it doesn't really help improve build times through parallelism or something (like on Travis).

Reply:
Ok.  I thought about moving this to its own job (as with Travis IIRC), but since the preceding steps only take a few minutes I think the gains would be small.

Reply:
This job also doesn't need to run unit tests and integration tests.

Moving it into a separate job makes sense -- the current build takes ~40 minutes and adding a separate parallel job where it'd probably spend ~20 minutes should help reduce the overall end-to-end time (which is what we should optimize IMO).

Reply:
> Do you just mean splitting unit and integration tests into separate jobs?

Yes. That's what I meant.

The thought was that both the builds that are parallel so usually it saves us time. However, given that it takes about a minute, making it a separate job doesn't really have much merit.

Sorry for any confusion my thinking out loud caused.